File: JavaScript/M03_JavaScriptProgrammingBasics/L05_WhileLoop/Lab/P08_GraduationII.js
function graduationPt2(input) {
    let name = input[0];

    let index = 1, grade = 4, sum = 0;
    while (grade >= 4 && index <= 12) {
        let grade = input[index];
        sum += parseFloat(grade);
        if (grade < 4) {
            console.log(`${name} has been excluded at ${index} grade`); break;
        }
        else if (index == 12) {
            let average = sum / 12;
            console.log(`${name} graduated. Average grade: ${average.toFixed(2)}`);
        }
        index++;
    }
}

File: C++/M01_ProgrammingBasics/L05_WhileLoop/Lab/Solutions/P08_GraduationII.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	string name, grade;
	cin >> name >> grade;

	int counter = 1;
	double currGrade = stod(grade);
	double avgGrade = currGrade;
	double sum = currGrade;
	while (4 <= avgGrade) {
		cin >> grade;
		currGrade = stod(grade);
		sum += currGrade;
		counter++;

		if (currGrade < 4) {
			break;
		}

		avgGrade = sum / counter;

		if (counter == 12) {
			cout.setf(ios::fixed);
			cout.precision(2);
			cout << name << " graduated. Average grade: " << avgGrade << endl;

			return 0;
		}
	}

	cout << name << " has been excluded at " << counter << " grade" << endl;

	return 0;
}
